### PR TITLE
Reimplements medic-reporter in admin app

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Adds a page which delivers the core functionality of medic-reporter
to the admin app which means sending messages and reports can be
tested easily and without needing a separate app.

Also removes the ability to set the locale via the reports api which
matches real world usage.

medic/medic-webapp#4516

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

**Comment by garethbowen** [comment]
> If we're dropping setting the locale via a versioned api, should that version not change?

Yeah that's a good call. The `put` turned to a `post` because we're not setting the _id so the `put` failed, which makes me think nobody is using this api... But just in case I'll make a v2.

File: api/src/server-utils.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ module.exports = {
     if (typeof err === 'string') {
       return module.exports.serverError(err, req, res);
     }
-    var code = err.code || err.statusCode || 500;
+    var code = err.code || err.statusCode || err.status || 500;
```
Comment (SCdF) [inline api/src/server-utils.js]: Lord above

File: admin/src/templates/message_test.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<form action="" method="POST" class="col-xs-8 configuration-form">
+
+  <div class="form-group required" ng-class="{'has-error': errors.message}">
+    <label for="message" translate><translate>Message Body</label>
```
Comment (SCdF) [inline admin/src/templates/message_test.html]: This doesn't look valid. What is the `<translate>` element, why do we need that and the translate attribute, and why does it not get closed?

Also, for new translations are we not using dot notation?

File: api/src/routing.js [hunk 3]
```diff
@@ -209,23 +209,13 @@ app.all(`/${db.getPath()}/export/:type/:form?`, exportData.routeV1);
...
-  });
-});
+app.post('/api/v1/records', [jsonParser, formParser], records.v1);
+app.post('/api/v2/records', [jsonParser, formParser], records.v2);
 
 app.get('/api/v1/forms', function(req, res) {
   forms.listForms(req.headers, function(err, body, headers) {
     if (err) {
-      return serverUtils.serverError(err, req, res);
+      return serverUtils.error(err, req, res);
```
Comment (SCdF) [inline api/src/routing.js]: did you mean to change this at the same time?

File: admin/src/templates/message_test.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    <input id="message" type="text" class="form-control" ng-model="message" />
+    <span class="help-block" ng-if="errors.message" translate>{{errors.message}}</span>
+  </div>
+
+  <div class="form-group required" ng-class="{'has-error': errors.from}">
+    <label for="from" translate><translate>from</label>
+    <input id="from" type="text" class="form-control small" ng-model="from" />
+    <span class="help-block" ng-if="errors.from" translate>{{errors.from}}</span>
```
Comment (SCdF) [inline admin/src/templates/message_test.html]: Does this one really get translated?

File: admin/src/templates/message_test.html [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<form action="" method="POST" class="col-xs-8 configuration-form">
+
+  <div class="form-group required" ng-class="{'has-error': errors.message}">
+    <label for="message" translate><translate>Message Body</label>
+    <input id="message" type="text" class="form-control" ng-model="message" />
+    <span class="help-block" ng-if="errors.message" translate>{{errors.message}}</span>
```
Comment (SCdF) [inline admin/src/templates/message_test.html]: Does this one really get translated?

File: api/src/server-utils.js (hunk 1 above)
Comment (garethbowen) [reply api/src/server-utils.js]: Yeah. Types? Anyone?

File: admin/src/templates/message_test.html (hunk 2 above)
Comment (garethbowen) [reply admin/src/templates/message_test.html]: Yeah. Um... autocomplete fail I guess?

Comment (garethbowen) [reply admin/src/templates/message_test.html]: Fixed.

File: api/src/routing.js (hunk 3 above)
Comment (garethbowen) [reply api/src/routing.js]: Yeah, I noticed it was also wrong while I was in there, fixing the records api.

File: admin/src/templates/message_test.html (hunk 4 above)
Comment (garethbowen) [reply admin/src/templates/message_test.html]: Yeah. First the angular expression is resolved to a translation key, then it's translated.

File: admin/src/templates/message_test.html (hunk 5 above)
Comment (garethbowen) [reply admin/src/templates/message_test.html]: As above.

File: admin/src/templates/message_test.html (hunk 2 above)
Comment (garethbowen) [reply admin/src/templates/message_test.html]: This isn't a new translation it's reusing an old one.

**Pull request merged**